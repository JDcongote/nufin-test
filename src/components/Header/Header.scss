@import '../../styles/variables/index';
@import'../../styles/tools/mixins';

.nav-header {
  width: 100%;
  height: 40px;
  background-color: var(--color--background-secondary);
  position: relative;
  display: flex;


  &__title {
    font-weight: bolder;
    text-transform: uppercase;
    letter-spacing: 0.2em;
    font-size: 20px;
    margin: auto;
    align-self: center;
    color: var(--color--secondary);

    @include breakpoint(phone-up) {
      margin: 0 1rem;
      flex: 1;
    }
  }

  &__hamburger {
    position: absolute;
    cursor: pointer;
    right: 5%;
    align-self: center;
    z-index: var(--z-index--level-3);
    -webkit-tap-highlight-color: transparent;

    .line {
      width: 30px;
      height: 3px;
      background: #fff;
      margin: 5px;
      transition: transform 100ms;

      .rotator {
        width: 100%;
        height: 100%;
        background-color: #fff;
      }
    }

    &--opened {
      .line:nth-child(1) {
        display: none;
      }

      .rotator {
        transform: rotateZ(90deg);
        transform-origin: center center;
      }

      .line:nth-child(2) {
        transform: rotateZ(-45deg);
        transform-origin: center center;
      }

      .line:nth-child(3) {
        display: none;
      }
    }

    @include breakpoint(phone-up) {
      display: none;
      pointer-events: none;
    }
  }

  &__menu {
    position: fixed;
    list-style: none;
    width: 100%;
    height: 100vh;
    clip-path: circle(0px at 90% 3%);
    transition: all 200ms ease-out;
    background: var(--color--primary);
    color: var(--color--secondary);
    pointer-events: none;
    z-index: var(--z-index--level-2);
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    padding: 10px;

    &--opened {
      clip-path: circle(200% at 90% 3%);
      pointer-events: all;
    }

    &-item {
      font-size: 2em;
      font-weight: bold;

      @include breakpoint(phone-up) {
        font-size: 1em;
        font-weight: 600;
        ;
      }
    }

    @include breakpoint(phone-up) {
      position: relative;
      display: inline;
      width: auto;
      height: auto;
      display: flex;
      flex-direction: row;
      clip-path: none;
      background: transparent;
      pointer-events: all;
      justify-content: space-around;
      width: 50%;
    }
  }
}