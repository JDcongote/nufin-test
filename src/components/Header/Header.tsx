import React, { useState } from 'react';
import './Header.scss';

type HeaderProps = {
  items: any[];
  onMenuClick: (item: any) => void;
  pageTitle: string;
  currentPage?: string;
};

const Header = (props: HeaderProps) => {
  const [menuOpen, openMenu] = useState(false);

  return (
    <header>
      <nav className="nav-header">
        <h1 className="nav-header__title">{props.pageTitle}</h1>

        <div
          className={
            'nav-header__hamburger' +
            (menuOpen ? ' nav-header__hamburger--opened' : '')
          }
          onClick={() => openMenu(!menuOpen)}
        >
          <div className="line"></div>
          <div className="line">
            <div className="rotator"></div>
          </div>
          <div className="line"></div>
        </div>
        <ul
          className={
            'nav-header__menu' + (menuOpen ? ' nav-header__menu--opened' : '')
          }
        >
          {props.items.map(item => (
            <li
              key={item.id}
              className={
                item.id === props.currentPage
                  ? 'nav-header__menu-item nav-header__menu-item--active'
                  : 'nav-header__menu-item'
              }
              onClick={() => {
                props.onMenuClick(item);
                openMenu(false);
              }}
            >
              {item.name}
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
};

export default Header;
