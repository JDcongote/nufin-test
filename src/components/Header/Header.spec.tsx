import Header from './Header';
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Header', () => {
  it('Basic Header Test', () => {
    const items = [{ id: '1', name: 'a' }, { id: '2', name: 'b' }];
    const pageTitle = 'TITLE';
    const onMenuClick = jest.fn();

    const wrapper = mount(
      <Header items={items} onMenuClick={onMenuClick} pageTitle={pageTitle} />
    );

    const render = renderer.create(
      <Header items={items} onMenuClick={onMenuClick} pageTitle={pageTitle} />
    );

    expect(wrapper.find('.nav-header__title').text()).toBe(pageTitle);
    expect(
      wrapper
        .find('.nav-header__menu-item')
        .first()
        .text()
    ).toBe('a');

    expect(render.toJSON()).toMatchSnapshot();
  });
});
