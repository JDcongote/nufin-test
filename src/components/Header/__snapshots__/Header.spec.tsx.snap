// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Basic Header Test 1`] = `
<header>
  <nav
    className="nav-header"
  >
    <h1
      className="nav-header__title"
    >
      TITLE
    </h1>
    <div
      className="nav-header__hamburger"
      onClick={[Function]}
    >
      <div
        className="line"
      />
      <div
        className="line"
      >
        <div
          className="rotator"
        />
      </div>
      <div
        className="line"
      />
    </div>
    <ul
      className="nav-header__menu"
    >
      <li
        className="nav-header__menu-item"
        onClick={[Function]}
      >
        a
      </li>
      <li
        className="nav-header__menu-item"
        onClick={[Function]}
      >
        b
      </li>
    </ul>
  </nav>
</header>
`;
